cluster_configs/barnacle/barnacle_status.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python
+
+import subprocess
+import sys
+import time
+import re
+from io import StringIO
+
+def parse_qstat(lines):
+    status = {}
+
+    for line in lines:
+        if line.startswith(' '):
+            entry = line.rstrip()
+            p = re.compile('^ +(.+?) = (.+)')
+            m = p.match(entry.rstrip())
+            status[m.groups()[0]] = m.groups()[1]
+        elif line.startswith('\t'):
+            entry += line.strip()
+            p = re.compile('^ +(.+?) = (.+)')
+            m = p.match(entry.rstrip())
+            status[m.groups()[0]] = m.groups()[1]
+
+    return(status)
+
+def print_status(status):
+    try:
+        if status['job_state'] == 'R':
+            print("running")
+        elif status['job_state'] == 'Q':
+            print("running")
+        elif status['job_state'] == 'C' and status['exit_status'] == '0':
+            print("success")
+        elif status['job_state'] == 'C' and status['exit_status'] != '0':
+            print("failed")
+        else:
+            print("failed")
+    except KeyError:
+        print("failed")
+
+def main():
```

Comment:
Looks like there is lots of non-standard things here: `sys.argv` instead of `click`, absence of docstring, hard-coded constants, absence of unit test. I am totally fine with it though. Just a reminder.

Reply:
this is a helper script for Snakemake and so doesn't interact with user---it has very prescribed inputs (one positional argument) and outputs (one of 'success', 'running', or 'failure'), so I'd rather rely on on standard libraries than Click. It is specific to Barnacle, but mostly follows [this example](https://github.com/Snakemake-Profiles/surfsara-grid/blob/master/%7B%7Bcookiecutter.profile_name%7D%7D/grid-status.py) from Johannes Köster. 

But the unit test thing is true, I do need to do that.

Reply:
Makes sense!